cmd/thanos/tools_bucket.go

```diff
@@ -377,9 +380,16 @@ func registerBucketLs(app extkingpin.AppClause, objStoreConfig *extflag.PathOrCo
 			return err
 		}
 
-		fetcher, err := block.NewMetaFetcher(logger, block.FetcherConcurrency, bkt, "", extprom.WrapRegistererWithPrefix(extpromPrefix, reg), nil, nil)
-		if err != nil {
-			return err
+		var fetcher *block.MetaFetcher
+
+		if tbc.excludeDelete {
+			ignoreDeletionMarkFilter := block.NewIgnoreDeletionMarkFilter(logger, bkt, 0, block.FetcherConcurrency)
+			fetcher, err = block.NewMetaFetcher(logger, block.FetcherConcurrency, bkt, "", extprom.WrapRegistererWithPrefix(extpromPrefix, reg), []block.MetadataFilter{ignoreDeletionMarkFilter}, nil)
```

Comment:
This can be extracted out of the if else block.
You can just use something like:
``` go
var filters []block.MetadataFilter
if tbc.excludeDelete {
    // add ignore deletion filter to filters
}
fetcher, err = block.NewMetaFetcher(xxx)
```

Reply:
Thanks! Fixed in a recent commit.